collective/dexteritytextindexer/indexer.py

```diff
@@ -33,6 +34,12 @@ def __init__(self, context, request):
 def dynamic_searchable_text_indexer(obj):
     """Dynamic searchable text indexer.
     """
+    # if the object does not provide a request, get one.
+    # This happens when running scripts (bin/instance run script.py)
+    try:
+        obj.REQUEST
+    except AttributeError:
+        obj.REQUEST = getRequest()
```

Comment:
@jone my first approach was like that, but then I realized that ``obj.REQUEST`` is being used in multiple places, so I thought about making a minimal patch.

I will change the patch to always use getRequest.